### ISSUE TITLE
Crash on Android 14 (S24 Ultra) when setting edges prop on RNCSafeAreaView

### ISSUE THREAD

**Issue by carloslibardo** [th3rdwave/react-native-safe-area-context#475]
### Description

I've encountered a crash in my React Native application on Android when trying to set the `edges` prop on `RNCSafeAreaView`. The application crashes with the following exception:

Exception com.facebook.react.bridge.JSApplicationIllegalArgumentException: Error while updating property 'edges' of a view managed by: RNCSafeAreaView
...
Caused by java.lang.ClassCastException: com.facebook.react.bridge.ReadableNativeMap cannot be cast to com.facebook.react.bridge.ReadableArray

This issue seems to be related to how the `edges` prop is handled in the native Android implementation of `RNCSafeAreaView`.

### Steps to Reproduce

1. Use `SafeAreaView` with the `edges` prop set to an array of strings, e.g., `edges={['top', 'left', 'right']}`.
2. Run the app on an Android device or emulator.

### Expected Behavior

The `SafeAreaView` should correctly apply the safe area insets according to the specified edges without causing a crash.

### Actual Behavior

The application crashes on Android with a `JSApplicationIllegalArgumentException` related to the `edges` prop.

### Environment

- React Native version: 0.73.4
- React Native Safe Area Context version: 4.9.0
- Device: S24 Ultra (I cannot reproduce in my device - S23 Ultra)

### Code Snippet

Here are the snippets from my code where I use `SafeAreaView`:

```jsx
<SafeAreaView
  edges={['top', 'left', 'right']}
  style={{ flex: 1 }}
>
...
```
### Additional Context
I've followed the documentation for react-native-safe-area-context and the usage of SafeAreaView seems to be correct according to the examples provided. This crash is blocking further development and testing on Android devices.

Any assistance or guidance on how to resolve this issue would be greatly appreciated.

**Comment by jacobp100** [comment]
Are you able to get a better stack trace?

**Comment by carloslibardo** [comment]
For now I don't think so, this is the Google Play stacktrace, I even have Sentry configured, but for some reason it is not registering this crash.

Furthermore, I can't reproduce the error locally as I don't have the Samsung Galaxy S23 at hand, this is the log of a customer with problems in production.

**Comment by jacobp100** [comment]
Are you able to reproduce it on a device?

**Comment by carloslibardo** [comment]
No, because I don't have the s24 ultra, what I'm managing to get is a problem from a customer in production who is helping us with some tests.

**Comment by jacobp100** [comment]
I’ll leave this issue open - with the caveat it’s unlikely to get taken on by any current maintainers. If you find more information - or better, a fix - let us know!

**Comment by kuldeep-IT** [comment]
Stucked in same issue.
